### ISSUE TITLE
Topics

### ISSUE THREAD

**Issue by richorama** [OrleansContrib/meetups#1]
Thread to discuss topics you would like to cover. Please also suggest talks you would like to give.

**Comment by sergeybykov** [comment]
Should we do the roadmap/Q&A discussion next week? How about Monday late morning PST, so that it's not too late in Europe and not too early in Australia? Will this work for you guys?

**Comment by ReubenBond** [comment]
Works for me

-----Original Message-----

**Comment by yevhen** [comment]
Ok for me

**Comment by jkonecki** [comment]
I should make it as well.

**Comment by johnazariah** [comment]
Yeah that'll work for me!

**Comment by richorama** [comment]
Great, how about 11am PST, which is 7pm GMT and 6am on Tuesday in Melbourne: http://www.worldtimebuddy.com/?qm=1&lid=8,2158177,2643743&h=8&date=2015-1-12&sln=11-12

**Comment by sergeybykov** [comment]
Sounds good to me. If it's not too early for AU folks. Should we do it in Google Hangout again? I've never used it before. Am I reading it right that it's limited to 10 concurrent users?

**Comment by davidgristwood** [comment]
I am away skiing next week - I hope to catch up with folk when I get back...

**Comment by sergeybykov** [comment]
Oh, I see that Richard already create one. Thanks!

**Comment by richorama** [comment]
Great. I've set up a hangout, I'll invite you to it later on. 10 is the limit for presenters, the number of viewers is unlimited. I'll also add the folks contributing on this thread as presenters. 

@David enjoy the skiing!

-----Original Message-----

**Comment by sergeybykov** [comment]
Great. That should be plenty. :-)

**Comment by richorama** [comment]
You (should) be able to join the hangout as a presenter using this link: https://plus.google.com/hangouts/_/hoaevent/AP36tYfbH6alRA_M7xLFZ5i9Lj5RNiMjrgziNftchl6cOKCJTq4hHA

**Comment by sergeybykov** [comment]
Yes, I can get in. Thanks!

**Comment by richorama** [comment]
I'd like to hear a talk on the new streaming grain. Any volunteers?

@yevhen have you got a date for your talk, or shall I press on with something else first?

**Comment by sergeybykov** [comment]
@richorama Do you mean the streaming functionality/APIs?

**Comment by richorama** [comment]
@sergeybykov correct.

**Comment by sergeybykov** [comment]
We could do an intro to the streaming functionality in the next meetup after @yevhen's. Docs should also be available by then.

**Comment by richorama** [comment]
that would be great!

**Comment by sergeybykov** [comment]
In about a month after @evhen's talk, which we still don't have a date set for?

**Comment by yevhen** [comment]
Hey, guys!

Sorry for being silent for so long. Can hardly find spare time to work on prezi (with all that pre-war situation in my country). I'm dedicating any free time I can find for work on uniform API for Orleans. Basically, it's done. Any further polish will probably be better handled by community.

I'll try to find some time next week and get back to you.

Sorry,
Yevhen

**Comment by yevhen** [comment]
What about Friday, 6 of March?

**Comment by richorama** [comment]
Sounds good to me! Same time (19:00 UTC)?

-----Original Message-----

**Comment by davidgristwood** [comment]
cool!

meanwhile, still interested to know about Azure Event Hub with Orleans

**Comment by sergeybykov** [comment]
@davidgristwood, why did you close this issue? :-) I thought it was meant to stay open for future meetups.

**Comment by richorama** [comment]
I'm assuming David did this by accident

**Comment by davidgristwood** [comment]
ooops hit wrong button - sorry

**Comment by yevhen** [comment]
@richorama 
> Sounds good to me! Same time (19:00 UTC)?

Ok for me

**Comment by richorama** [comment]
Link to get into the hangout (in 5 minutes): https://plus.google.com/hangouts/_/hoaevent/AP36tYeOyeir0H7veQEzMFzdTZCPDnyP6iyPZwCO2QJyJ-yvZxrUrw

**Comment by yevhen** [comment]
Slides https://t.co/arfhU8njAj

2015-03-06 20:53 GMT+02:00 Richard Astbury <notifications@github.com>:

**Comment by yevhen** [comment]
Link to repo https://github.com/yevhen/Orleankka
Nuget https://www.nuget.org/packages/Orleankka
We'll introduce it next week.Also update nuget package with the recent developments.

**Comment by ReubenBond** [comment]
Thanks, @yevhen - sorry I didn't make it :( Reading through it all now

**Comment by yevhen** [comment]
@ReubenBond no problem The video is on youtube already. Might be better than silent slides. Also, discussion at the end was valuable. Didn't have time to present the [project](https://github.com/yevhen/Orleankka). Maybe next meetup :)

**Comment by ReubenBond** [comment]
Very interesting, @yevhen. You lose compile-time type safety, though - doesn't that bother you? What about deriving your grain contract from multiple `IEventHandler<TEvent>` interfaces - one for each event - instead of using `object` and `dynamic`. You get to keep type safety and don't suffer the runtime penalty of `dynamic` (https://stackoverflow.com/questions/13193799/performance-cost-of-using-dynamic-typing-in-net).

**Comment by yevhen** [comment]
@ReubenBond 
> You lose compile-time type safety, though - doesn't that bother you?

Not at all. I do exhaustive unit testing :)
Even if you don't - it wasn't such a big problem before.

Akka does a similar thing with its `TypedActor` base class and `IHandles<T>' interface. But it's only for invocation side. The point here was that there is no need to create a dedicated grain contract interface, since the set of messages (along with exceptions) which grain could process/raise, fully defines it's public interface (contract).

**Comment by johnazariah** [comment]
hahah - I think Yehven meant to say "Yes, I do lose compile-time type
safety, but I cover for it with unit testing" :)

I raised this as a comment as well - and I think Yehven agreed that the F#
DU model has stronger compile-time guarantees :)

On Sat, Mar 7, 2015 at 9:13 AM, Yevhen Bobrov <notifications@github.com>
wrote:

> @ReubenBond <https://github.com/ReubenBond>
>
> You lose compile-time type safety, though - doesn't that bother you?
>
> Not at all. I do exhaustive unit testing :)
> Even if you don't - it wasn't such a big problem before.
>
> Akka does a similar thing with its TypedActor base class and `IHandles'
> interface. But it's only for invocation side. The point here was that there
> is no need to create a dedicated grain contract interface, since the set of
> messages (along with exceptions) which grain could process/raise, fully
> defines it's public interface (contract).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OrleansContrib/meetups/issues/1#issuecomment-77646030>
> .
>



-- 
- John

*John Azariah* | Microsoft Azure MVP
<https://mvp.microsoft.com/en-us/mvp/John%20Azariah-5000463>



*m *+61 421 100 747
*e* john.azariah@gmail.com
*t* @johnazariah <http://twitter.com/johnazariah>

**Comment by yevhen** [comment]
@ReubenBond 
> You get to keep type safety and don't suffer the runtime penalty of dynamic (https://stackoverflow.com/questions/13193799/performance-cost-of-using-dynamic-typing-in-net)

That was just an example. Something simple. just to showcase the concept. In fact, I don't use `dynamic` much but rather complied lambda expressions, which is pretty close to a direct method call.

Also, the information in this SO post is outdated. `dynamic` was highly optimized in recent versions of .NET framework  (same as `Activator.CreateInstance`) so it's usually only slow on a first call and then all further invocations are incredibly fast.

**Comment by yevhen** [comment]
@johnazariah 
> hahah - I think Yehven meant to say "Yes, I do lose compile-time type safety, but I cover for it with unit testing" :)

Indeed :)

**Comment by ReubenBond** [comment]
That's fair enough, Yehven. I guess I like the discoverability which static typing gives me, letting me autocomplete my way around.

**Comment by yevhen** [comment]
@ReubenBond 
> I guess I like the discoverability which static typing gives me, letting me autocomplete my way around

Is that the only thing that might stop you from trying this approach (message passing). Compared to all other advantages that you might get when using it? :)

How many messages per actor type you now have, so that the absence of auto-discoverability is a real issue?

**Comment by yevhen** [comment]
Also, I can show you the trick how to make auto-discoverability possible with message passing. It requires creating an additional extension method for uniform interface and some trivial inheritance modeling skills  :)

It's basically just a way to represent DUs in C# ...

**Comment by yevhen** [comment]
@johnazariah 
>  I raised this as a comment as well - and I think Yehven agreed that the F# DU model has stronger compile-time guarantees :)

Yes, this approach (message passing) looks really sleek in F#. Example from my PoC https://github.com/yevhen/Orleankka/blob/master/Source/FSharp.Demo/Shop.fs

That should change with C#7 when record types and pattern matching will be introduced

**Comment by ReubenBond** [comment]
I've used this approach before (in Akka & in one of my early Orleans projects). Message passing is great, but I prefer explicit contracts - Orleans uses message passing under the hood, anyway.

If I were to go down this route, I would likely use this approach:
```
public interface IMyGrain : IGrain, IHandleEvent<EventOne>, IHandleEvent<EventTwo>{}
```

Where IHandleEvent<T> is just
```
public interface IHandleEvent<T>
{
    Task Handle(T @event);
}
```

**Comment by yevhen** [comment]
@ReubenBond
Yes, unfortunately it's not quite the same. That will make an interface non-uniform and you will lose a higher-order function, through which each and every message will be delivered to a grain. And it is the most important thing here.

But if you don't need the power that such a higher-order receive function could give, then even the current Orleans' method-based approach will be fine.

**Comment by yevhen** [comment]
And all of this is problematic only due to C# FP immaturity. With F# - message passing is superior :)

**Comment by ReubenBond** [comment]
Why would you lose the ability to use higher-order functions? Instead of `object`, my functions operate on `TEvent`. Do you have an example? I'm not saying C#'s type system is particularly robust, but it's robust enough for most cases.

**Comment by yevhen** [comment]
Because Orleans will dispatch `TEvent` directly to a concrete handler, the one that has Handle(TEvent e).

**Comment by johnazariah** [comment]
Hey I'm loving what I'm seeing of Orleankka! Can I start using it? :D

**Comment by yevhen** [comment]
@johnazariah sure :)
I'll be updating Nuget package next week with recent developments.
It should be completely on-par with all of the features of native Orelans API.

What is missing is:
1. Automatic deactivation timeouts one per actor-type basis (1 hour of work even for newbie, with uniform interface approach)
2. Fluent configuration for Azure - if you've seen what we did there with simplified configuration DSL, then you know what I'm talking about :)

Thanks for compliment!

**Comment by yevhen** [comment]
@johnazariah But these polishing should not introduce any breaking changes. We might change signature of `IActorActivator` but that should be a single place change for clients, so nothing very serious here

**Comment by johnazariah** [comment]
Nice - i can't contribute right now, but in 2-3 months, I'll have some time, hopefully!

**Comment by yevhen** [comment]
@johnazariah cool! We're thinking about moving it to OrleansContrib.

**Comment by johnazariah** [comment]
OK - that sounds great!

**Comment by richorama** [comment]
Who is up for speaking at the next meetup? @gabikliot mentioned that he could talk about the streaming API, but @ReubenBond is also lined up for talking about how he's using Orleans at some point? Do you guys have any dates you could provide?

**Comment by gabikliot** [comment]
If @ReubenBond can do the next one, I think it will be preferred from our side. We will the do one after that and hopefully by that time we will be done with stream provider refactoring, so there will be more to talk about. But if Reuben can't, we can do the next one as well. How about meetup #4 in 3 weeks and meetup #4 a month later?

**Comment by ReubenBond** [comment]
I can do the next one, whenever you would like to schedule it, at least 2 weeks from now.

I'll talk about "How we use Orleans in FreeBay", if people are interested.
I'll write up some slides and whatnot, but I'll cover AuthN/AuthZ, the admin console, the streaming API, and how the system is modeled.

If people have suggestions for things they would like me to cover, I will incorporate them.

**Comment by johnazariah** [comment]
+1 for the streaming API

On Thu, Mar 12, 2015 at 2:05 PM, Reuben Bond <notifications@github.com>
wrote:

> I can do the next one, whenever you would like to schedule it, at least 2
> weeks from now.
>
> I'll talk about "How we use Orleans in FreeBay", if people are interested.
> I'll write up some slides and whatnot, but I'll cover AuthN/AuthZ, the
> admin console, the streaming API, and how the system is modeled.
>
> If people have suggestions for things they would like me to cover, I will
> incorporate them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OrleansContrib/meetups/issues/1#issuecomment-78416063>
> .
>



-- 
- John

*John Azariah* | Microsoft Azure MVP
<https://mvp.microsoft.com/en-us/mvp/John%20Azariah-5000463>



*m *+61 421 100 747
*e* john.azariah@gmail.com
*t* @johnazariah <http://twitter.com/johnazariah>

**Comment by richorama** [comment]
@ReubenBond how does Friday 10th April 19:00 UTC sound?

**Comment by ReubenBond** [comment]
Looks good to me. Lock it in :) thanks 

-----Original Message-----

**Comment by ReubenBond** [comment]
Something has come up, could we please shift this by a couple of days? Any time in the week following is suitable for me.

**Comment by richorama** [comment]
No probs. Do you want to update the readme with a new date? I'm OOF at the moment...

-----Original Message-----

**Comment by ReubenBond** [comment]
Updated to Tue, 15th April 2015, 19:00 - 20:00 GMT. Does that work? I don't have the rights to update the Google Hangout event, are you able to update that?

**Comment by richorama** [comment]
Hmm, I might struggle to make it (I think it's the only day that week which I'm out on the road). Although I'm not essential for being there!

**Comment by ReubenBond** [comment]
Let's bump it to Wednesday, then :) I've updated the README.

**Comment by richorama** [comment]
:-)

-----Original Message-----

**Comment by richorama** [comment]
Link for the current meetup - starting minutes...
https://plus.google.com/hangouts/_/hoaevent/AP36tYfV4VTQggg4M1DDYLy6ggGhwFvRyAj-tZA9gPrlZFl_Rlolew

**Comment by sergeybykov** [comment]
We don't have a date set up for next meetup where @gabikliot is going to dive into streaming, do we?

**Comment by richorama** [comment]
Not yet ;-) do you have have a date Gabi?

-----Original Message-----

**Comment by gabikliot** [comment]
How about 3 weeks from now, May 15th or May 22nd?

**Comment by javier-alvarez** [comment]
I'm  interested and day should work for me

-----Original Message-----

**Comment by richorama** [comment]
@gabikliot either date is great.

It would be great if someone else could volunteer to host the meetup, there's a good chance I won't be able to join as I'm moving house (to a house with no internet at the moment) and having a baby (well my wife is - but I need to help apparently).

**Comment by sergeybykov** [comment]
@richorama Congratulations with the baby!

I've never hosted a Hangout, but if nobody more experienced steps in, I can try to play Richard this time.

**Comment by gabikliot** [comment]
Indeed, congratulations with the baby Richard!
I think lets pick May 22nd? It will give us enough time to finish the stream providers refactoring we are working on now and present a more coherent story.

**Comment by richorama** [comment]
Brilliant, let's make it the 22nd, and if Sergey could host, that would be great!

-----Original Message-----

**Comment by sergeybykov** [comment]
By 22nd I'll beat jetlag and hopefully won't fall asleep at the Hangout wheel. Richard, if there are any instructions I need to know, please send them my way.

**Comment by sergeybykov** [comment]
19:00 GMT, right?

**Comment by richorama** [comment]
Any time you like. I normally go for 7pm GMT (well BST at the moment)

-----Original Message-----

**Comment by richorama** [comment]
I'll write up what I do to organise a meet up.

-----Original Message-----

**Comment by javier-alvarez** [comment]
Can you send a calendar invite for this?

**Comment by richorama** [comment]
I think the google hangout creates one.

-----Original Message-----

**Comment by sergeybykov** [comment]
We are going live in 45 minutes at https://plus.google.com/events/crdjm977pqubv81lgdhqruoqal4.

**Comment by yevhen** [comment]
@sergeybykov crouching ... :smile:

**Comment by sergeybykov** [comment]
@yevhen Beer-o-clock already? ;-)

**Comment by yevhen** [comment]
@sergeybykov gym-o-clock :) Actually stuck in a traffic jam going back to home ... but I should in time

**Comment by NeverOddOrEven** [comment]
"richorama commented on Dec 23, 2014
3. Failure modes in Orleans, and what to expect (i.e. network partitions, failed nodes)."

I have been trying to answer questions at my organization surrounding different error modalities. Unfortunately, I am not finding the resources I would like that cover the topic with the depth I need. For instance: 

1. If a grain serializes data out to a local hard drive, and the hosting silo goes offline forever, have I lost that state forever? Is there an expectation on the infrastructure that the file system is highly available and durable (SAN, Azure Table, etc...)? 

2. If a message is delivered to a grain, and the hosting silo dies before the task completes an asynchronous process, is that considered a message delivery failure, or does my application logic need to handle that case with some task timeout logic?

I am sure there are plenty of other unique cases prevalent in distributed systems, but those two have been nagging me the most. A conversation about "do this, not that" would be informative.

**Comment by sergeybykov** [comment]
1. It is in general not recommended to persist any data you don't want to lose on a local drive simply because there is always a chance of a fatal hardware failure. With high-throughput and relatively low-latency remote storage solutions, be it SAN or SQL/table/blob stores, the recommended option is to use them for persisting grain state.

2. Every call to a grain returns a Task that will be broken in case of a failure, either right away (in a failure happened before or during sending of the request) or when the call timeout (by default 30 seconds) expires. Application code needs to await the Task and handle errors.

**Comment by NeverOddOrEven** [comment]
Wow, fast response! I believe error modalities would still be a great google hangout, but this answers my pressing questions. Thank you.

**Comment by sergeybykov** [comment]
Great!

BTW, we don't have our next meetup scheduled yet. Any volunteers to present or suggestions what topic(s) you'd like to cover?

**Comment by meliketotest** [comment]
I would love to see something about graphs in Orleans!

**Comment by richorama** [comment]
Sounds good, anyone using Orleans in an interesting way, which they could share at the next meet up?

-----Original Message-----

**Comment by gabikliot** [comment]
Re graphs, we build project Horton in the past. We (the Microsoft Orleans team) can definitely share our experience. I don't think we have enough content to fill a meet up. It might be better just to open an issue on GH about it, and we can share our approach and experience. Of course if someone else has anything interesting to share, that could be cool.

**Comment by sergeybykov** [comment]
I spoke with BigPark guys about them showing https://github.com/OrleansContrib/OrleansTemplates in action and talking about some other interesting components they are building for easier deployment of Orleans-based services. They are happy to present in ~2 weeks time frame. Does this sound interesting?

**Comment by jkonecki** [comment]
Yes, sure!

Speaking about deployments maybe @reubenbond could spend 5 mins talking
about running Orleans in Fabric Services as well....

**Comment by richorama** [comment]
Sounds great. I'm happy to set up and host...

-----Original Message-----

**Comment by richorama** [comment]
hey, @nehmebilal - are you up for a meetup? How does 8th July at 19:00 GMT work for you?

**Comment by gabikliot** [comment]
Both, me and Sergey are busy on July 8th and 10th.
Can we move it to the next week - July 13-17?

**Comment by richorama** [comment]
ok, how about 17th July?

**Comment by richorama** [comment]
I haven't got any contact details for @nehmebilal could someone put me in touch?

**Comment by sergeybykov** [comment]
@richorama I DM'ed you Nehme's email.

**Comment by nehmebilal** [comment]
Hi Richard,

Sorry for late reply!

I am a bit unclear on what the meetup is about? Is it to present OrleansTemplates?

Thanks,
Nehme

**Comment by richorama** [comment]
Thanks Sergey!

Neheme, if your willing to do a talk on OrleansTemplates that would be great (no pressure). Or if there's something else you'd prefer to cover that could also be interesting....

-----Original Message-----

**Comment by nehmebilal** [comment]
There is potentially something else (deployment story) but we're not sure when we can present that. Maybe it's better to wait until we have the deployment story ready?

**Comment by richorama** [comment]
Ok, how about this - perhaps the next meetup @ReubenBond and @nehmebilal could do a slot each (on Orleans on the Service Fabric, and OrleansTemplates). (I know Reuben hasn't offered - but I'm sure if we speak nicely to him he'd be up for it :¬)

Then when @nehmebilal has the deployment story ready, we could have that as a future meetup?

**Comment by nehmebilal** [comment]
Sure, that works.

**Comment by richorama** [comment]
I'm postponing the meetup. Nehme is waiting for some internal sign-off so he can show up some more stuff!

**Comment by sergeybykov** [comment]
We ran a Special Intern Project (four PhD interns) with Microsoft Research this summer with aimed at adding support for geo-distribution to Orleans. We are planning to start bringing that functionality to GitHub in the near future. I thought it would be useful to do a meetup about that whole projects, its approaches, and algorithms before we start bringing PRs with the code.

If this sounds interesting we could do it next Friday, October 23rd.

**Comment by jkonecki** [comment]
That would be awesome.

**Comment by richorama** [comment]
Sounds great, let's make it happen. Is the usual time ok?

**Comment by sergeybykov** [comment]
11am PST is okay for Sebastian and Phil that will be doing the presentation.

**Comment by yevhen** [comment]
Cool! Won't be able to join but will definitely watch recorded. I' have birthday party this day :smile:

**Comment by yevhen** [comment]
btw, if there is interest I can talk next about all cool things in Orleankka?

**Comment by sergeybykov** [comment]
@yevhen If you have a change of heart, our geek party will be there for you. :-) 

I think it'll be great to hear an update on Orleankka. We need to start borrowing some ideas back.

**Comment by richorama** [comment]
@yevhen cool, it would be interesting to get an update. Could you suggest a date when you get a change to look at your diary?

**Comment by jkonecki** [comment]
@yevhen sure - we will sing happy birthday for you ;-)

**Comment by yevhen** [comment]
I'll be probably helping @AntyaDev to prepare his presentation for upcoming MS DevDay (http://devday.in.ua/) conference, where he'll be talking about Orleans/Orleankka. So, I'll have something already cooked ;) 

I think 3 weeks from the next meetup will be a good date for me. But let's return to this topic in few weeks :smile:

**Comment by ReubenBond** [comment]
+1 for @yevhen talking about Orleankka :)
Excited to hear about geo-distribution!

**Comment by AntyaDev** [comment]
+ 1 for @yevhen

**Comment by ghuntley** [comment]
+1 for @yevhen as well.

**Comment by sergeybykov** [comment]
Next week is a short one in US because of Thanksgiving. Should we schedule the next meetup for Friday December 4th?

**Comment by sergeybykov** [comment]
We could discuss the roadmap or have @yevhen present Orleankka if he is ready and has time.

**Comment by yevhen** [comment]
Hi guys,

Ye, I'll be happy to present Orleankka. But 4th of December is too soon ((
If we can do it at least a week later, then I'll have enough time to
prepare.

Yevhen

2015-11-23 5:59 GMT+02:00 Sergey Bykov <notifications@github.com>:

**Comment by richorama** [comment]
@yevhen cool, could you pick a date which works for you?

**Comment by yevhen** [comment]
Dec 17 will work for me.

2015-11-24 22:01 GMT+02:00 Richard Astbury <notifications@github.com>:

**Comment by jkonecki** [comment]
An early Christmas present ;-)

**Comment by Johnny-Bee** [comment]
I'll be there!

**Comment by sergeybykov** [comment]
Looking forward to it.

What's a good date for discussing the roadmap then? After the New Year?

**Comment by jkonecki** [comment]
Sure: new year, new resolutions, new challenges.... ;-)

**Comment by richorama** [comment]
It would be great to have a roadmap meetup in the new year. Is there a date that the core team could do?

**Comment by ReubenBond** [comment]
@sergeybykov are you able to make time for a roadmap meetup this month?

**Comment by sergeybykov** [comment]
I was about to suggest Thursday January 21st. Is the usual 11 PST/19 GMT time still best for most people?

**Comment by johnazariah** [comment]
works for me!

On Wed, Jan 6, 2016 at 8:51 AM, Sergey Bykov <notifications@github.com>
wrote:

> I was about to suggest Thursday January 21st. Is the usual 11 PST/19 GMT
> time still best for most people?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OrleansContrib/meetups/issues/1#issuecomment-169385673>
> .
>



-- 
- John

*John Azariah*

*m *+1 425 614 8158
*e* john.azariah@gmail.com
*t* @johnazariah <http://twitter.com/johnazariah>

**Comment by galvesribeiro** [comment]
Its ok for me too ;)

**Comment by richorama** [comment]
Sounds good, I'll set up the hangout

**Comment by richorama** [comment]
done: https://plus.google.com/u/0/events/cdjfo4lb86ong99re802e5mfso8

**Comment by richorama** [comment]
It would be really interesting to have a YAMS meetup, @nehmebilal would you be prepared to present?

**Comment by jkonecki** [comment]
+1
Could @reubenbond show Service Fabric deployments at the same time?

**Comment by ReubenBond** [comment]
I'm using Yams at the moment, @jkonecki - [differential uploads via YamsDeploy](https://github.com/ReubenBond/YamsDeploy) make it much faster for me than I can feasibly achieve with Fabric at the moment. If people would like to see Orleans on SF, I can demo that.

**Comment by jkonecki** [comment]
That would be nice, I think, to see different deployment options.

**Comment by sergeybykov** [comment]
:+1: [Balmer's voice] Deployment! Deployment! Deployment!

**Comment by jkonecki** [comment]
:-D

**Comment by nehmebilal** [comment]
Sounds good! But when is the next meetup? :)

**Comment by richorama** [comment]
Whenever you like! Usually one eventing at 7pm.

**Comment by nehmebilal** [comment]
Perfect, I'll get back to you :)

**Comment by sergeybykov** [comment]
Just to be clear - 7pm GMT / 11am PST is the usual time slot.

**Comment by nehmebilal** [comment]
Feb 16, 19, 22, 26 or later? @ReubenBond?

**Comment by ReubenBond** [comment]
Feb 22, 26, & later work for me.

**Comment by nehmebilal** [comment]
Feb 26?

**Comment by richorama** [comment]
Great, let's do 26th Feb 19:00 UTC

I'll set up the hangout in the next day or so.

I look forward to it!

**Comment by richorama** [comment]
Would anyone but up for presenting on a show dedicated to telemetry and monitoring?

As discussed with @sergeybykov on gitter it could cover:

1. how folks do telemetry integration, monitoring, and alerting today; 
1. what we need to do going forward to a) make the experience smoother and less error prone; b) remove duplication of functionality within the codebase and overlaps with external tooling; c) cross-platform aspects.

**Comment by richorama** [comment]
It would be great to do meetup themed on deploying and updating Orleans. Does anyone have an interesting strategy for this that they can share?

**Comment by jkonecki** [comment]
I've just got my custom Octopus Deploy step template for YAMS included into Octopus Library. I could briefly explain how we host Orleans using Yams and deploy using Octopus - I would imagine 10 mins will suffice.

**Comment by richorama** [comment]
@jkonecki sounds great. 

Would be particularly interested in:

* How you handle changes to persisted state (i.e. to ensure that state is backward compatible).
* How do you roll back?
* What impact does it have on system performance?
* What are the areas of pain?

**Comment by yevhen** [comment]
@jkonecki how do you notify clients so they can reconnect (restore stream subscriptions)?

**Comment by sergeybykov** [comment]
When should we do the meetup? Thursday December 8th?

**Comment by richorama** [comment]
@sergeybykov good idea, before people get too busy with the festive period. @jkonecki are you available?

**Comment by jkonecki** [comment]
Yes, 8th Dec is fine for me

**Comment by sergeybykov** [comment]
What time will work reasonably for AU-EU-US? 7 MEL-SYD / 12 PST / 20 BST?

**Comment by richorama** [comment]
http://www.worldtimebuddy.com/?qm=1&lid=2643743,5809844,2158177&h=2643743&date=2016-12-8&sln=20-21

I think it's a good time (I don't see any objection!). I'll update the readme.

**Comment by richorama** [comment]
Looking for a topic for the next meetup, if anyone is interested?

One suggestion I have is a 'deep dive on Orleans internals'.

**Comment by ElanHasson** [comment]
I like that topic.

**Comment by yevhen** [comment]
FSM could be the other choice. @johnazariah can guide us through his new shiny project, @jkonecki may share experience with Stateless, I can talk about switchable behaviors in Orleankka.

**Comment by johnazariah** [comment]
Happy to!!!

**Comment by jkonecki** [comment]
I can show how we're using Stateless + Orleankka + Events - won't take more
than 5 mins though

Jakub

**Comment by richorama** [comment]
Would people find a mailing list useful, to announce meetups?

**Comment by jkonecki** [comment]
Yes. Also other announcements like releases / conferences / presentations /
blog posts.

**Comment by galvesribeiro** [comment]
I think Mailing lists get things lost specially if you deal with a big amount of emails... Would be better if you create an Issue with an announcement, just like asp.net people do. Whoever is subscribed will be notified.

**Comment by sergeybykov** [comment]
@galvesribeiro You mean a single issue in dotnet/orleans? I like that better than a mailing list.

**Comment by galvesribeiro** [comment]
@sergeybykov I meant something like this https://github.com/aspnet/Announcements

Whenever they want to publish an announcement, everyone gets notified. This repo has just it. All issues are closed etc. But they have their own organization, Orleans doesn't. So I suggest create a repo like that here in OrleansContrib org just for that. So all blog post, release, breaking change get notified here.

**Comment by richorama** [comment]
I think that's a good idea.

**Comment by richorama** [comment]
I have created an announcements repo: https://github.com/OrleansContrib/Announcements
